state/machine.go

```diff
@@ -2322,6 +2379,21 @@ func (m *Machine) UpdateOperation() *UpdateMachineOperation {
 	return &UpdateMachineOperation{m: &Machine{st: m.st, doc: m.doc}}
 }
 
+func (m *Machine) getUpgradeSeriesLock() (*upgradeSeriesLockDoc, error) {
```

Comment:
let's use the new getUpgradeSeriesLock() in machine SetUpgradeSeriesStatus() and UpgradeSeriesStatus() too.

Reply:
I think methods like `isUpgradeSeriesComplete` and higher level patterns might be better. In truth I think `isUpgradeSeriesComplete` is too `ask` rather than `tell`. I think we can come up with some nice methods here rather than keep on asking for a lock to examine. We might be able to move methods onto a lock object if nice patterns emerge.

Reply:
Can we add a TODO in SetUpgradeSeriesStatus() and UpgradeSeriesStatus()?

Reply:
Lets wait on this